### PR TITLE
Add xweb based REST API for the fabric

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: controller/api_impl/base_entity_api_model.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
+/*
```
Comment (andrewpmartinez) [inline controller/api_impl/base_entity_api_model.go:1]: Mmmmm again lots of code duped from the edge. Link builders and other base entity items should be consolidated over time.

File: controller/api_impl/router_router.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		return wrapper.WrapRequest(func(n *network.Network, rc api.RequestContext) { r.Create(n, rc, params) }, params.HTTPRequest, "", "")
+	})
+
+	//fabricApi.RouterUpdateRouterHandler = router.UpdateRouterHandlerFunc(func(params router.UpdateRouterParams) middleware.Responder {
+	//	return wrapper.WrapRequest(func(n *network.Network, rc api.RequestContext) { r.Update(n, rc, params) }, params.HTTPRequest, params.ID, "")
+	//})
+	//
+	//fabricApi.RouterPatchRouterHandler = router.PatchRouterHandlerFunc(func(params router.PatchRouterParams) middleware.Responder {
+	//	return wrapper.WrapRequest(func(n *network.Network, rc api.RequestContext) { r.Patch(n, rc, params) }, params.HTTPRequest, params.ID, "")
+	//})
```
Comment (andrewpmartinez) [inline controller/api_impl/router_router.go:66]: ?

File: specs/swagger.yml [hunk 3]
```diff
@@ -0,0 +1,1397 @@
...
+paths:
+  ###################################################################
+  # Services
+  ##################################################################
+  '/services':
+    get:
+      summary: List services
+      description: |
+        Retrieves a list of service resources; supports filtering, sorting, and pagination. Requires admin access.
+      tags:
```
Comment (andrewpmartinez) [inline specs/swagger.yml:30]: What kind of auth is required? Just a client cert?

File: specs/swagger.yml [hunk 4]
```diff
@@ -0,0 +1,1397 @@
...
+  ##################################################################
+  '/links':
+    get:
+      summary: List links
+      description: |
+        Retrieves a list of link resources; doesn not supports filtering, sorting, or pagination. Requires admin access.
+      tags:
+        - Link
+      operationId: listLinks
+      #      parameters:
```
Comment (andrewpmartinez) [inline specs/swagger.yml:427]: Documentation says it support filter/sort/pagination - but inputs are commented out

**Review by andrewpmartinez — APPROVED** [review]
See comments, minor changes

File: controller/api_impl/base_entity_api_model.go (hunk 1 above)
Comment (plorenz) [reply controller/api_impl/base_entity_api_model.go:1]: Agree. I did a first pass at reducing duplication, but there's lots left to do

File: controller/api_impl/router_router.go (hunk 2 above)
Comment (plorenz) [reply controller/api_impl/router_router.go]: ziti-fabric doesn't allow updating routers. I've implemented it now.

File: specs/swagger.yml (hunk 3 above)
Comment (plorenz) [reply specs/swagger.yml:30]: Yes, when in pure fabric mode only a valid client cert is required (like current ziti-fabric). When running with the edge it piggy backs on edge authentication and requires a valid api session. 

I was concerned about removing the security section, but from I can see in the edge code we just ignore that parameter and enforce the presence of a valid ztSession in our code, so I thought it was ok.

File: specs/swagger.yml (hunk 4 above)
Comment (plorenz) [reply specs/swagger.yml]: Doc does it does NOT support those things (with a misspelling). Fix misspelling and removed comment out inputs

File: specs/swagger.yml (hunk 3 above)
Comment (andrewpmartinez) [reply specs/swagger.yml:30]: Open API 2.0 / Swagger doesn't have a way to specify client cert auth. The only option you have is to document it on every endpoint with words.

Comment (andrewpmartinez) [reply specs/swagger.yml:30]: It is interesting that this API has two security modes though. I'm not sure how to document it w/o having two specs.

**Pull request merged**